src/main/java/net/dries007/tfc/CommonEventHandler.java

```diff
@@ -68,6 +77,44 @@ public static void onBlockHarvestDrops(BlockEvent.HarvestDropsEvent event)
         }
     }
 
+    @SubscribeEvent
+    public static void onRightClickWater(PlayerInteractEvent event)
+    {
+        if (event.getSide() == Side.CLIENT) return;
+        if (!(event instanceof PlayerInteractEvent.RightClickBlock || event instanceof PlayerInteractEvent.RightClickEmpty))
+            return;
+        EntityPlayer player = event.getEntityPlayer();
+        if (!(player.getHeldItemMainhand().isEmpty() && player.getHeldItemOffhand().isEmpty()))
+            return; //Need both hands to drink water
```

Comment:
well, it's already harder to drink water using both your hands. Using only one should be even harder, that's why i made impossible, but can be tweaked to drink half the amount or ignore offhand entirely.

Reply:
I see your point, but I'm trying to see this from a gameplay perspective. There's no real "two handed" actions in minecraft (I think the only one is the crossbow, which has a real visual indicator that your offhand is not in use. Even so, you can still have something in the offhand). I think this would just be a point of player frustration - I for one basically always have torches in my offhand and having to switch them out every time I want to drink would be a pain

Reply:
Well, to further increase this discussion, this is only for drinking directly from water sources. You can, and should, use containers for that. Glass bottles, Water jugs and such should be used instead of drinking using your hands. It's waaaay easier irl, and at least IMO for TFC perspective, should be a mechanic.